packages/at_auth/lib/src/at_auth_impl.dart

```diff
@@ -115,8 +126,22 @@ class AtAuthImpl implements AtAuth {
           ' and try again (or) contact support@atsign.com');
     }
     //2. generate key pairs
-    _atAuthKeys = BaseAtKeysIo.generateKeyPairs(
-        atOnboardingRequest.authMode, publicKeyId: atOnboardingRequest.publicKeyId);
+    AtKeysIo atKeysIo = FileAtKeysIo();
+
+    switch (atKeysIo) {
+      // case SimAtKeysIo():
+      //   if (publicKeyId == null || publicKeyId.isEmpty) {
+      //     throw AtAuthenticationException('sim publicKeyId is required for sim auth mode');
+      //   }
+      //   _atAuthKeys = atKeysIo.generateKeys(publicKeyId);
+      //   break;
+      case FileAtKeysIo():
+        _atAuthKeys = atKeysIo.generateKeyPairs();
+        await atKeysIo.write(atOnboardingRequest.atSign, _atAuthKeys);
+        break;
+      default:
+        throw AtAuthenticationException('Unsupported AtKeysIO implementation: ${atKeysIo.runtimeType}');
+    }
```

Comment:
We definitely have to do a sealed class here, this won't be extendable otherwise